File: lib-segment/src/main/java/websiteschema/mpsegment/filter/SegmentResultFilter.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package websiteschema.mpsegment.filter;

import websiteschema.mpsegment.conf.MPSegmentConfiguration;
import websiteschema.mpsegment.core.SegmentResult;

import java.util.ArrayList;

/**
 * @author ray
 */
public class SegmentResultFilter {

    private final ArrayList<ISegmentFilter> filters;

    public SegmentResultFilter(MPSegmentConfiguration config) {
        filters = new ArrayList();
        filters.add(new UnknownPlaceFilter());
        filters.add(new NumberAndTimeFilter());
        filters.add(new UnknownNameFilter(config));
        filters.add(new ReduplicatingFilter());
        filters.add(new QuerySyntaxFilter(config));
        if (config.is("segment.lang.en")) {
            filters.add(new EnglishStemFilter());
        }
    }

    public void filter(SegmentResult segmentResult) {
        for (ISegmentFilter filter : filters) {
            filter.setSegmentResult(segmentResult);
            filter.filtering();
        }
    }
}

File: lib-segment-scala/src/main/scala/websiteschema/mpsegment/filter/SegmentResultFilter.scala
package websiteschema.mpsegment.filter

import ner.{ForeignNameRecognizerCreator, ChineseNameRecognizerCreator}
import websiteschema.mpsegment.conf.MPSegmentConfiguration
import websiteschema.mpsegment.core.SegmentResult

import collection.mutable.ListBuffer

class SegmentResultFilter(config: MPSegmentConfiguration) {

  private val filters = ListBuffer[ISegmentFilter]()
  filters += (new UnknownPlaceFilter())
  filters += (new NumberAndTimeFilter())

  if (config.isChineseNameIdentify()) {
    if (config.getNameRecognizer == "UnknownNameFilter")
      filters += (new UnknownNameFilter(config))
    if (config.getNameRecognizer == "ChineseNameFilter")
      filters += (new ChineseNameFilter(config, new ChineseNameRecognizerCreator(), 10))
  }

  filters += (new ReduplicatingFilter())
  filters += (new QuerySyntaxFilter(config))

  if (config.is("segment.lang.en")) {
    filters += (new EnglishStemFilter(config.is("segment.lang.en.stemming")))
  }

  if (config.isHalfShapeAll() || config.isUpperCaseAll()) {
    filters += (new UpperCaseAndHalfShapeFilter(config.isHalfShapeAll(), config.isUpperCaseAll()))
  }

  def filter(segmentResult: SegmentResult) {
    for (filter <- filters) {
      filter.setSegmentResult(segmentResult)
      filter.filtering()
    }
  }
}
